### ISSUE TITLE
Support for ReturnCount Rule to exclude return count for guard clauses preceded by require and check functions

### ISSUE THREAD

**Issue by amitdash291** [detekt/detekt#1910]
## Expected Behavior of the rule
ReturnCount rule has an option to ignore use of guard clauses as the first statement of a function, it should also ignore such guard clauses even when they are preceded by `require` and `check` functions.

## Context
Essentially if `require` and `check` functions precede a guard clause then the guard clause should not affect return count. A sample code block is shared below:

```kotlin
fun testFun(testVal: String?) {
  require(testVal != null)
  check(testVal != null)
  if (testVal != null) return
}
```
For the above function the return expression `if (testVal != null) return` should not be counted by the ReturnCount rule.

**Comment by schalkms** [comment]
In addition to Artur's comment we should think about whether it's worth adding complex checks to this (simple) rule for the mentioned use case.

**Comment by amitdash291** [comment]
@arturbosch @schalkms how should we proceed with this, should we do the enhancement in the way Artur described in the comment above or not do this at all?

**Comment by schalkms** [comment]
I'd say so. I'd wait for Artur's reply please.

**Comment by arturbosch** [comment]
Let's do it ^^. I think it is a nice addition for the rule.

**Comment by amitdash291** [comment]
@schalkms @arturbosch Please assign this to me, I will try to complete this.

**Comment by schalkms** [comment]
@amitdash291 done

**Comment by amitdash291** [comment]
@arturbosch this issue seems to be fixed as a part of #2342  
please check and close this issue if the above is correct
cc @schalkms

**Comment by schalkms** [comment]
Closed because the mentioned case is not reproducible.
